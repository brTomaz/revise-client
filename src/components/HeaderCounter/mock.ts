import { HeaderCounterProps } from '.'

export const mockHeaderCounter: HeaderCounterProps = {
  topText: 'Pontuação',
  mainInfo: 60,
  bottomText: 'pontos'
}
