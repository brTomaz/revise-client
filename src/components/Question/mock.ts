/* eslint-disable @typescript-eslint/no-empty-function */
export const mockQuestion = {
  id: 1,
  info:
    '<figure class="image"><img src="/uploads/question_1_bca75dd891.png" alt=""></figure>',
  statement:
    'No cartum, a crítica está no fato de a sociedade exigir do adolescente que ',
  answers: [
    'se aposente prematuramente.',
    'amadureça prematuramente.',
    'estude aplicadamente.',
    'se forme rapidamente.',
    'ouça atentamente.'
  ],
  correctAnswer: 2,
  handleClick: () => {}
}
