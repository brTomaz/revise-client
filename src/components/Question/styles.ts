import styled, { css, keyframes } from 'styled-components'
import media from 'styled-media-query'

import { Wrapper as AnswerWrapper } from 'components/Answer/styles'

const fadeInUpBig = keyframes`
   from {
    opacity: 0;
    transform: translate3d(0, 2000px, 0);
  }

  to {
    opacity: 1;
    transform: translate3d(0, 0, 0);
  }
`

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${media.greaterThan('medium')`
      box-shadow: 0 0 0.4rem 0.5rem 0 rgba(0, 0, 0, 0.2);
      padding: 0 ${theme.spacings.xxlarge};
    `}
  `}
  animation: ${fadeInUpBig} 0.6s ease;
`

export const InfoWrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.darkGray};
    border-radius: ${theme.border.radius} ${theme.border.radius}
      ${theme.border.radius} 0;
    padding: ${theme.spacings.small} ${theme.spacings.small};
    margin-bottom: ${theme.spacings.xlarge};

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.small} ${theme.spacings.xlarge};
    `}

    p {
      font-size: ${theme.font.sizes.large};
      line-height: 1.4;
      text-align: justify;

      max-width: 100%;

      ${media.greaterThan('medium')`
        font-size: ${theme.font.sizes.xlarge};
      `}
    }

    img {
      max-width: 100%;
    }

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  `}
`

export const StatementAnswerWrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: 0.4rem;
    border-radius: ${theme.border.radius};

    ${AnswerWrapper} {
      margin-bottom: 0.4rem;
      :last-child {
        border-radius: 0 0 2.4rem 2.4rem;
        margin-bottom: 0;
      }
    }
  `}
`
