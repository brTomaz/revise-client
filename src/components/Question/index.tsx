import { MouseEvent } from 'react'

import Answer from 'components/Answer'
import Statement from 'components/Statement'
import { TypeQuestion } from 'domain/entities/question'
import { HandleAnswerClickType } from 'domain/functions/handle-answer-click-type'
import * as Styles from './styles'

export type QuestionProps = Omit<TypeQuestion, 'id'> & {
  handleClick: HandleAnswerClickType
}

const Question = ({
  info,
  statement,
  answers,
  correctAnswer,
  handleClick
}: QuestionProps) => {
  return (
    <Styles.Wrapper>
      <Styles.InfoWrapper>
        <div dangerouslySetInnerHTML={{ __html: info }} />
      </Styles.InfoWrapper>
      <Styles.StatementAnswerWrapper>
        <Statement>{statement}</Statement>
        <ul>
          {answers?.map((answer, index) => {
            return (
              <Answer
                key={answer}
                handleAnswerClick={(event: MouseEvent) =>
                  handleClick(index, correctAnswer, event)
                }
                answer={answer}
              />
            )
          })}
        </ul>
      </Styles.StatementAnswerWrapper>
    </Styles.Wrapper>
  )
}

export default Question
