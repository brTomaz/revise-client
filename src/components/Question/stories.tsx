import { Story, Meta } from '@storybook/react/types-6-0'

import Question, { QuestionProps } from '.'
import { mockQuestion } from './mock'

export default {
  title: 'Question',
  component: Question,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story<QuestionProps> = (args) => <Question {...args} />

Default.args = mockQuestion
