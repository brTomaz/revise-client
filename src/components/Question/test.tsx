import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'
import { mockQuestion } from './mock'
import Question from '.'

describe('<Question />', () => {
  it('should render with statement and answers', () => {
    const { container } = renderWithTheme(<Question {...mockQuestion} />)

    const statement = screen.getByText(/defende o/i)
    const firstAnswer = screen.getByText(/aponta a/i)
    const lastAnswer = screen.getByText(/demonstra acreditar/i)

    expect(statement).toBeInTheDocument()
    expect(firstAnswer).toBeInTheDocument()
    expect(lastAnswer).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
