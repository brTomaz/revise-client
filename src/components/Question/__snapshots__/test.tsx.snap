// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> should render with statement and answers 1`] = `
.c5 {
  cursor: pointer;
  list-style: none;
  background: #040923;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  color: #FAFAFA;
  font-weight: 600;
  border-radius: 4px;
  padding: 1.6rem 1.6rem;
  font-size: 1.8rem;
}

.c5:hover {
  color: #2DC2C3;
}

.c3 {
  color: #030617;
  font-size: 1.8rem;
  font-weight: 400;
  padding: 1.6rem;
}

.c0 {
  margin-top: -3.6rem;
  min-width: 100%;
  -webkit-animation: eyJzjA 2.3s ease-in-out;
  animation: eyJzjA 2.3s ease-in-out;
}

.c1 {
  background-color: #FAFAFA;
  color: #3A3A3A;
  border-radius: 2.6rem 2.6rem 2.6rem 0;
  padding: 2.4rem 2.4rem;
  margin-bottom: 4.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 p {
  font-size: 1.8rem;
  line-height: 1.4;
  text-align: justify;
  max-width: 100%;
}

.c1 img {
  max-width: 100%;
}

.c2 {
  background-color: #FAFAFA;
  padding: 0.4rem;
  border-radius: 2.6rem;
}

.c2 .c4 {
  margin-bottom: 0.4rem;
}

.c2 .c4:last-child {
  border-radius: 0 0 2.4rem 2.4rem;
  margin-bottom: 0;
}

@media (min-width:768px) {
  .c5 {
    font-size: 2.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.0rem;
  }
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
    padding: 0 5.6rem;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 2.4rem 4.8rem;
  }
}

@media (min-width:768px) {
  .c1 p {
    font-size: 2.0rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <figure
          class="image"
        >
          <img
            alt=""
            src="https://res.cloudinary.com/dg64nrwjl/image/upload/v1608203495/question_4_c08561bb4d.png"
          />
        </figure>
      </div>
    </div>
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Ao abordar o assunto “obesidade”, em uma seção de jornal, o autor
      </h2>
      <ul>
        <li
          class="c4 c5"
        >
          defende o consumo liberado de açúcar.
        </li>
        <li
          class="c4 c5"
        >
          aponta a gordura como o grande vilão da saúde.
        </li>
        <li
          class="c4 c5"
        >
          demonstra acreditar que a obesidade não é preocupante.
        </li>
        <li
          class="c4 c5"
        >
          indica a necessidade de mais pesquisas sobre o assunto.
        </li>
        <li
          class="c4 c5"
        >
          enfatiza a redução de ingestão de calorias pelos americanos.
        </li>
      </ul>
    </div>
  </div>
</div>
`
